master/internal/telemetry/telemetry.go

```diff
@@ -61,32 +65,94 @@ func New(
 	}
 
-	return &TelemetryActor{db, rm, client, clusterID}, nil
+	syslog := logrus.WithFields(logrus.Fields{
+		"component":  fmt.Sprintf("telemetry"),
+		"clusterID":  clusterID,
+		"segmentKey": segmentKey,
+	})
+
+	return &Telemeter{
+		db:        db,
+		rm:        rm,
+		client:    client,
+		clusterID: clusterID,
+		syslog:    syslog,
+	}, nil
 }
 
-// Receive implements the actor.Actor interface.
-func (s *TelemetryActor) Receive(ctx *actor.Context) error {
-	switch msg := ctx.Message().(type) {
-	case actor.PreStart:
-		actors.NotifyAfter(ctx, 0, telemetryTick{})
+// Init sets up the Telemetry singleton.
+func Init(
+	system *actor.System,
+	db db.DB,
+	rm telemetryRPFetcher,
+	clusterID string,
+	conf config.TelemetryConfig,
+) {
+	if DefaultTelemeter != nil {
+		logrus.Warn(`detected re-initialization of Telemetry singleton `,
+			`that should never occur outside of tests`)
+		return
+	}
 
-	case analytics.Track:
-		msg.UserId = s.clusterID
-		if err := s.client.Enqueue(msg); err != nil {
-			ctx.Log().WithError(err).Warnf("failed to enqueue track %s", msg.Event)
-		}
+	if !conf.Enabled || conf.SegmentMasterKey == "" {
+		logrus.Info("telemetry reporting is disabled")
+		return
+	}
 
-	case telemetryTick:
-		// Tick in a random interval.
-		//nolint:gosec // Weak RNG is fine here.
-		randNum := rand.Intn(maxTickIntervalMins-minTickIntervalMins) + minTickIntervalMins
-		actors.NotifyAfter(ctx, time.Duration(randNum)*time.Minute, telemetryTick{})
+	telemetryDef, err := New(
+		db,
+		rm,
+		clusterID,
+		conf.SegmentMasterKey,
+	)
+	if err != nil {
+		logrus.WithError(err).Errorf("failed to initialize telemetry")
+		return
+	}
 
-		ReportMasterTick(ctx.Self().System(), s.db, s.rm)
+	DefaultTelemeter = telemetryDef
+	DefaultTelemeter.syslog.Info(`telemetry reporting is enabled; `,
+		`run with --telemetry-enabled=false to disable`)
+	go DefaultTelemeter.tick(system)
+}
 
-	case actor.PostStop:
-		_ = s.client.Close()
+// track adds track call objects to the analytics.Client interface.
+func (s *Telemeter) track(t analytics.Track) {
+	if s == nil {
+		return
 	}
 
-	return nil
+	t.UserId = s.clusterID
+	if err := s.client.Enqueue(t); err != nil {
+		s.syslog.WithError(err).WithField("event", t.Event).Warn("failed to enqueue track")
+	}
+}
+
+func (s *Telemeter) tick(system *actor.System) {
+	if s == nil {
+		return
+	}
+
+	for {
+		resp, err := s.rm.GetResourcePools(system, &apiv1.GetResourcePoolsRequest{})
+		if err != nil {
+			// TODO(Brad): Make this routine more accepting of failures.
+			s.syslog.WithError(err).Error("failed to receive resource pool telemetry information")
+			return
+		}
+		ReportMasterTick(resp, s.db)
+		time.Sleep(s.sleepInterval())
+	}
+}
+
+func (s *Telemeter) sleepInterval() time.Duration {
+	bg := big.NewInt(maxTickIntervalMins - minTickIntervalMins)
+	randNum, err := rand.Int(rand.Reader, bg)
+	if err != nil {
+		s.syslog.Error(err)
+		// Error handling: if random int can't be generated, return maxInterval
```

Comment:
i don't think this comment helps readability, personally. all it does it tell me the code is intentional, not a mistake.